sdk/core/azure-core/test/ut/context.cpp

```diff
@@ -18,7 +18,8 @@ TEST(Context, Basic)
   Context context;
   Context::Key const key;
 
-  EXPECT_FALSE(context.HasKey(key));
+  int placeholder;
+  EXPECT_FALSE(context.TryGetValue<int>(key, placeholder));
 }
 
 TEST(Context, BasicBool)
```

Comment:
I'd add a test case for `std::shared_ptr<something>` - the type that Storage stores there.

Reply:
Can you please provide an example code snippet of how to write such a test.

Reply:
Something like this? :

TEST(...)
{
    class S {
    private:
        int* m_instanceCount;
    public:
        S(int* instanceCount) : m_instanceCount(instanceCount) { ++(*m_instanceCount); }
        ~S() { --(*m_instanceCount); }
    };

    int instanceCount = 0;
    {
        auto sharedPtr = std::make_shared<S>(&instanceCount);
        EXPECT_EQ(sharedPtr.use_count(), 1);

        context = context.WithValue(key, sharedPtr);
        EXPECT_EQ(sharedPtr.use_count(), 2);

        std::shared_ptr<S> foundPtr;
        EXPECT_TRUE(context.TryGetValue(key, foundPtr));
        EXPECT_EQ(foundPtr.get(), sharedPtr.get());
        EXPECT_EQ(*instanceCount, 1);
        EXPECT_EQ(sharedPtr.use_count(), 1);
    }

   // Verify that context calls the destructor of shared_ptr it is holding
   EXPECT_EQ(*instanceCount, 0);
}
```

Reply:
storage is using shared ptr of `bool`, so I will use that, but also see if I can add the complex class scenario like you showed.